src/WidgetBase.ts

```diff
@@ -198,13 +191,14 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		this._properties = <P> {};
 		this._cachedChildrenMap = new Map<string | Promise<WidgetBaseConstructor> | WidgetBaseConstructor, WidgetCacheWrapper[]>();
 		this._diffPropertyFunctionMap = new Map<string, string>();
-		this._renderDecorators = new Set<string>();
 		this._bindFunctionPropertyMap = new WeakMap<(...args: any[]) => any, { boundFunc: (...args: any[]) => any, scope: any }>();
 		this._registries = new RegistryHandler();
 		this._registries.add(registry);
 		this.own(this._registries);
+		this._boundRenderFunc = this.render.bind(this);
```

Comment:
More of general observation of why we don't use initializers generally... like:

```ts
class WidgetBase {
    private _properties = {} as P;
    private _registries = new RegistryHandler();
}
```

It would cut out a fair amount of typing and is even more readable IMO.